src/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -244,16 +244,25 @@ TritonLoader::StartTriton()
   REPORT_TRITONSERVER_ERROR(
       api_version_fn_(&api_version_major, &api_version_minor),
       "unable to get api version");
-  if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
-      (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
+
+  if (TRITONSERVER_API_VERSION_MAJOR != api_version_major) {
```

Comment:
Add the case for where `TRITONSERVER_API_VERSION_MINOR` is > `api_version_minor`

We wanna fail in that case cuz PA is expecting a later minor version, meaning it could be using an API not found in the older minor version.

Reply:
Isn't it the other way round? When TRITONSERVER_API_VERSION_MINOR < api_version_minor, we fail it?

Reply:
`TRITONSERVER_API_VERSION_MINOR` is what PA is compiled to work with.

If it's higher than what it finds in the .so, it could mean it might use a not-yet-existing API.

That's why it's a failure for when `TRITONSERVER_API_VERSION_MINOR` > `api_version_minor`.

If you think this is true, can you provide reasoning?

Reply:
Harshini and I discussed offline, makes more sense from UX to only warn for minor mismatch, regardless of direction.

Reason being is that PA has been using the same Triton Core APIs since like 3 years ago when the feature was implemented. It's going to be highly unlikely for someone to be using a version of PA and Triton more than 3 years apart in version. I want to avoid the situation where someone has a server container a few months old and pip installs latest PA and gets an error because of that.